### PR TITLE
(WIP) Release process rehaul

### PR DESCRIPTION
**Pull request by frapposelli**
> DO NOT MERGE this PR, upstream work in `goreleaser` is not complete yet.

This PR intends to rehaul the release process for govc, adding vcsim to the list of artifacts published at every release, and keeping the door open for more (like `toolbox`).

Changes:
- Move to a single, autogenerated, changelog. the changelog creation is automated with `git-chlog` keeping the same standard structure as before. The old changelogs are preserved and point to the new one.
- `build.sh` and `release.sh` are removed, replaced by `goreleaser` and `git-chlog`.
- `RELEASE.md` was added, to explain and document the steps required to cut a govmomi release.
- `vcsim` was added as published artifact (binary, brew and docker image)
- Both `vcsim` and `govc` are compiled for `arm` and `arm64` platforms. 

/cc @dougm @akutz @lamw

### CODE REVIEW

File: vcsim/main.go [hunk 1]
```diff
@@ -42,6 +42,12 @@ import (
 	"github.com/vmware/govmomi/vim25/types"
 )
 
+// vcsim does not track versions like govc does, it only shows if a binary was
+// built during a tag event or from a git checkout.
+const Version = "dev"
```
Comment (dougm) [inline vcsim/main.go]: We could also move `govc/flags.Version` somewhere common and use it for both vcsim and govc.

File: CONTRIBUTORS [hunk 2]
```diff
@@ -27,15 +27,15 @@ bastienbc <bastien.barbe.creuly@gmail.com>
 Bob Killen <killen.bob@gmail.com>
 Brad Fitzpatrick <bradfitz@golang.org>
 Bruce Downs <bruceadowns@gmail.com>
-Cédric Blomart <cblomart@gmail.com>
```
Comment (dougm) [inline CONTRIBUTORS]: btw, this diff seems to be a difference with `sort` between Linux and Mac.

Comment (frapposelli) [reply CONTRIBUTORS]: yeah, I noticed as well, will investigate.

File: vcsim/main.go (hunk 1 above)
Comment (frapposelli) [reply vcsim/main.go]: good point, maybe it's time to move version to a top-level package

File: CONTRIBUTORS (hunk 2 above)
Comment (frapposelli) [reply CONTRIBUTORS]: I standardized `scripts/contributors.sh` on GNU sort, but running it on Mac does not change the outcome... interesting...

File: vcsim/main.go (hunk 1 above)
Comment (frapposelli) [reply vcsim/main.go]: made the move, now it's shared amongst all the packages (including toolbox)

**Comment by frapposelli** [comment]
Pushed an update with some improvements and added `toolbox` to the list of binaries built.

@akutz agree on your suggested changes, will make a separate commits as I need to take a second look at the current Makefile, I focused mostly on `goreleaser` as it is compiling the binaries independently, without calling the Makefile.

**Comment by embano1** [comment]
Closing this as #2340 and following PRs will fix it using Github Actions.

**Pull request closed without merging**